### PR TITLE
Full gimbal angle control

### PR DESCRIPTION
**Pull request by Daybreakerflint**
This feature is a request as I have to use the gimbal I have in a certain way and the roll and pitch axis are switched.
As the pitch of the payload of the gimbal is vitally important, this feature is needed.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
> and the roll and pitch axis are switched

Can you elaborate a bit? I am not sure I understand what this means... Are you saying that your Gimbal is reading the MAVLink message incorrectly, and therefore you want to invert the values you send from MAVSDK instead of fixing the Gimbal side?

**Comment by Daybreakerflint** [comment]
Hey Jonas!
I do not just want to invert the control, as my setup needs it because how i have it the gimbal is balanced and I have the least amount of weight. 

My payload on the gimbal is a quad rigged horn antenna. An antenna with two polarization. 
Because the drone payload is sensitive to such things as alignment. Meaning also how the polarization is to the object.
IMG_20240109_153559

The gimbal is reading everything ok. I just want full control of where it points to.

**Comment by JonasVautherin** [comment]
I see, so you need the roll axis. It was initially not exposed because cameras don't really need to set the roll axis :innocent:.

@julianoes: do you have an opinion on that?

File: protos/gimbal/gimbal.proto [hunk 1]
```diff
@@ -71,6 +80,15 @@ service GimbalService {
      rpc SubscribeControl(SubscribeControlRequest) returns(stream ControlResponse) {}
 }
 
+message SetAnglesRequest {
+    float pitch_deg = 1; // Pitch angle in degrees (negative points down)
+    float yaw_deg = 2; // Yaw angle in degrees (positive is clock-wise, range: -180 to 180 or 0 to 360)
+    float roll_deg = 3; // Roll angle in degrees
```
Comment (julianoes) [inline protos/gimbal/gimbal.proto:86]: Can we please change the order of the fields to match the usual MAVLink order: roll, pitch, yaw.

**Review by julianoes — CHANGES_REQUESTED** [review]

File: protos/gimbal/gimbal.proto (hunk 1 above)
Comment (Daybreakerflint) [reply protos/gimbal/gimbal.proto:86]: Sure. Shall I do it, or do you do it?  As I said, this is my first time. Happy to help where I can.

Comment (julianoes) [reply protos/gimbal/gimbal.proto:86]: I suggest you change it. Thanks!

**Review by JonasVautherin — APPROVED** [review]

**Review by julianoes — APPROVED** [review]

**Pull request merged**